awx/main/tasks/jobs.py

```
@@ -219,6 +227,53 @@ def __init__(self):
+            if src.source_credential.credential_type.name in WORKLOAD_IDENTITY_CREDENTIAL_TYPES
+        )
+        for credential_ctx, input_src in credential_input_sources:
+            if flag_enabled("FEATURE_OIDC_WORKLOAD_IDENTITY_ENABLED"):
+                try:
+                    jwt = retrieve_workload_identity_jwt(
+                        self.instance, audience=input_src.source_credential.get_input('url'), scope="aap_controller_automation_job"
+                    )
+                    # Set runtime context on credential (transient, not persisted to database)
+                    credential_ctx["workload_identity_token"] = jwt
+                    break  # Only need to set once per credential
+                except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)